scripts/cnode-helper-scripts/cncli.sh

```diff
@@ -127,11 +130,20 @@ getLedgerData() { # getNodeMetrics expected to have been already run
 }
 
 getConsensus() {
-  getProtocolParams
+  if isNumber "$1" ; then
+    if [[ "$1" == "$next_epoch" ]]; then
+       getProtocolParams || return 1
+    else
+       getProtocolParamsHist "$1" || return 1
+    fi
```

Comment:
I dont like to rely on some arbitrary variable that might or might not be set outside of function. Actually, leaderlogs for an epoch should always use parameters from the previous epoch to determine the consensus used. So we might as well always use -1. This was confirmed by Sam, that nonce calculation is done at the end of previous epoch, this its that epochs parameters that determine the consensus. 

New logic:
```
if isNumber "$1" ; then
  getProtocolParamsHist "$(( $1 - 1 ))" || return 1
else
  getProtocolParams || return 1
fi
```

Reply:
cool insight, thank you

Reply:
applied